airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -56,6 +58,18 @@
 USE_PER_RUN_LOG_ID = hasattr(DagRun, "get_log_template")
 
 
+VALID_ES_CONFIG_KEYS = set(inspect.signature(elasticsearch.Elasticsearch.__init__).parameters.keys())
```

Comment:
I've always don't know about such stuff in Top Level on the one side it should imported only once, in the other we might save for import time and move it into the `get_es_kwargs_from_config`

Reply:
The other things, since it method of the class inspect also include `self`. Possibly minor thing and could be ignored

Reply:
And also current implementation we also fix renamed parameter which not presented in elasticsearch 8

https://github.com/apache/airflow/blob/1bd321d07885d659ce91d7c3ab08764362da0093/airflow/providers/elasticsearch/log/es_task_handler.py#L104-L108

I don't know what we should exactly do with that parameter. IMO, if we yanked 5.0.0/5.0.1 then we could extend breaking changes and info that user should change `retry_timeout` to `retry_on_timeout` for keep previous behaviour

Reply:
> I've always don't know about such stuff in Top Level on the one side it should imported only once, in the other we might save for import time and move it into the get_es_kwargs_from_config

yeah. I think we are importing a lot more here and this inspect will be rather cheap comparing with importing the whole  es stack.

Reply:
> I don't know what we should exactly do with that parameter. IMO, if we yanked 5.0.0/5.0.1 then we could extend breaking changes and info that user should change retry_timeout to retry_on_timeout for keep previous behaviour

I included it in the `get_es_kwargs_from_config` method.

> The other things, since it method of the class inspect also include self. Possibly minor thing and could be ignored

Yep. Removed it and added test.